gapic/samplegen/samplegen.py

```diff
@@ -297,7 +324,49 @@ def validate_response(self, response):
 
             validater(self, body)
 
-    def _handle_lvalue(self, lval):
+    def validate_expression(self, exp: str) -> wrappers.Field:
+        """Validate an attribute chain expression.
+
+        Given a lookup expression, e.g. squid.clam.whelk,
```

Comment:
The expression could contain `[]`, `{}` and `%` as well. e.g., `$resp.books[0].name`.

Reply:
That answers a question I added to the features sheet, which is whether indexed lookups are permitted.

Reply:
Wait, how can the expression contain "%"?

Reply:
That's to access an entity of a resource name.

For example if the response contains a resource name location:

```
message FooResponse {

  // this string is a resource name, it follows the pattern "projects/{project}/locations/{location}"
  string location = 1;

}
```

Then

`$resp.location%project` refers to the `{project}` part of the `location` field.

The `%` is not needed for 1.2 though, so it's totally fine if you don't implement it for now. But we'll need to do this in the near future so just reserve the flexibility to support it later.

Reply:
That is not a thing I knew anything about. Is there documentation about it in the samplegen docs? Or documentation elsewhere?

Reply:
Sorry the doc missed this :( Tagged you where I updated the doc.

Reply:
I've added support for subscripting repeated attributes, e.g. `$resp.squids[0]`. I'm going to hold off on adding mapping lookup until a subsequent review, as I still have some work to do before supporting maps in general. I'll add a TODO.